panic on startup: flag redefined: home-disable-cluster-discovery (v7.1.31-0) — blocks codex system-role fix

## Summary

`v7.1.31-0` panics on startup with a duplicate flag-registration error, before it parses the config. The process exits immediately, so the release is completely unusable. It panics even on `--help`.

This is significant because `v7.1.31-0` ships the fix *"handle system role as developer in Claude request conversion"*, which is needed so the **codex (ChatGPT) provider** accepts Claude Code's system prompt. Without that fix (i.e. on `6.9.36-0` and earlier), routing Claude Code → codex returns `400 {"detail":"System messages are not allowed"}`. Because `7.1.31-0` cannot start, users are stuck without the fix.

## Environment

- **Plus version:** `7.1.31-0-plus` (Commit `a4addc55`, BuiltAt `2026-05-30T15:21:04Z`)
- **OS:** Windows 11 (amd64)
- **Go (build):** 1.26.3
- **Installed via:** CCS `v8.1.4`, `backend: plus` (`ccs cliproxy --install 7.1.31-0 --backend plus`)

## Panic / stack trace

```
CLIProxyAPI Version: 7.1.31-0-plus, Commit: a4addc55, BuiltAt: 2026-05-30T15:21:04Z
<path>\cli-proxy-api-plus.exe flag redefined: home-disable-cluster-discovery
panic: <path>\cli-proxy-api-plus.exe flag redefined: home-disable-cluster-discovery

goroutine 1 [running]:
flag.(*FlagSet).Var(0x..., {0x...}, {0x..., 0x1e}, {0x..., 0x54})
	/opt/hostedtoolcache/go/1.26.3/x64/src/flag/flag.go:1028 +0x385
flag.BoolVar(...)
	/opt/hostedtoolcache/go/1.26.3/x64/src/flag/flag.go:755
main.main()
	/home/runner/work/CLIProxyAPIPlus/CLIProxyAPIPlus/cmd/server/main.go:151 +0xd65
exit status 2
```

## Likely cause

The flag `home-disable-cluster-discovery` is registered twice via `flag.BoolVar` (around `cmd/server/main.go:151`). Go's `flag` package panics on a duplicate flag name during init, so the binary dies before doing anything.

## Steps to reproduce

```
# any of these panic immediately:
cli-proxy-api-plus.exe --help
cli-proxy-api-plus.exe --config <config.yaml>
ccs cliproxy --install 7.1.31-0 --backend plus && ccs cliproxy restart
```

`ccs cliproxy status` then shows `Not running` ("CLIProxy Plus failed to start within 5s on port 8317").

## Impact

- `7.1.31-0` is unusable on startup.
- It is the only release containing the codex system-role fix, so codex routing for Claude Code stays broken (`400 "System messages are not allowed"`) on the last working version (`6.9.36-0`).

## Workaround

Pin to `6.9.36-0` (`ccs cliproxy --install 6.9.36-0 --backend plus`) — it starts, but cannot route codex with the current Claude Code CLI.

## Request

A patched release (e.g. `7.1.31-1` / `7.1.32`) removing the duplicate `home-disable-cluster-discovery` flag registration would unblock the codex system-role fix. Thank you!

--- a/cmd/server/main.go
+++ b/cmd/server/main.go
@@ -148,7 +148,6 @@ func main() {
 	flag.StringVar(&password, "password", "", "")
 	flag.StringVar(&homeJWT, "home-jwt", "", "Home control plane JWT for mTLS certificate bootstrap and connection")
 	flag.BoolVar(&homeDisableClusterDiscovery, "home-disable-cluster-discovery", false, "Disable Home CLUSTER NODES discovery and keep using the configured Home address")
-	flag.BoolVar(&homeDisableClusterDiscovery, "home-disable-cluster-discovery", false, "Disable Home CLUSTER NODES discovery and keep using the configured -home-jwt address")
 	flag.BoolVar(&tuiMode, "tui", false, "Start with terminal management UI")
 	flag.BoolVar(&standalone, "standalone", false, "In TUI mode, start an embedded local server")
 	flag.BoolVar(&localModel, "local-model", false, "Use embedded model catalog only, skip remote model fetching")

--- a/cmd/server/main_flags_test.go
+++ b/cmd/server/main_flags_test.go
@@ -1,0 +1,85 @@
+package main
+
+import (
+	"go/ast"
+	"go/parser"
+	"go/token"
+	"strconv"
+	"testing"
+)
+
+func TestServerFlagsAreRegisteredOnce(t *testing.T) {
+	fset := token.NewFileSet()
+	file, err := parser.ParseFile(fset, "main.go", nil, 0)
+	if err != nil {
+		t.Fatalf("parse main.go: %v", err)
+	}
+
+	seen := make(map[string]token.Pos)
+	duplicates := make(map[string][2]token.Pos)
+
+	ast.Inspect(file, func(node ast.Node) bool {
+		call, ok := node.(*ast.CallExpr)
+		if !ok || len(call.Args) < 2 {
+			return true
+		}
+
+		selector, ok := call.Fun.(*ast.SelectorExpr)
+		if !ok || !isFlagRegistration(selector) {
+			return true
+		}
+
+		name, ok := stringLiteral(call.Args[1])
+		if !ok {
+			return true
+		}
+
+		if first, exists := seen[name]; exists {
+			duplicates[name] = [2]token.Pos{first, call.Args[1].Pos()}
+			return true
+		}
+		seen[name] = call.Args[1].Pos()
+		return true
+	})
+
+	if len(duplicates) == 0 {
+		return
+	}
+
+	for name, positions := range duplicates {
+		t.Errorf(
+			"flag %q registered more than once at %s and %s",
+			name,
+			fset.Position(positions[0]),
+			fset.Position(positions[1]),
+		)
+	}
+}
+
+func isFlagRegistration(selector *ast.SelectorExpr) bool {
+	ident, ok := selector.X.(*ast.Ident)
+	if !ok || ident.Name != "flag" {
+		return false
+	}
+
+	switch selector.Sel.Name {
+	case "BoolVar", "DurationVar", "Float64Var", "Func", "Int64Var", "IntVar",
+		"StringVar", "TextVar", "Uint64Var", "UintVar", "Var":
+		return true
+	default:
+		return false
+	}
+}
+
+func stringLiteral(expr ast.Expr) (string, bool) {
+	lit, ok := expr.(*ast.BasicLit)
+	if !ok || lit.Kind != token.STRING {
+		return "", false
+	}
+
+	value, err := strconv.Unquote(lit.Value)
+	if err != nil {
+		return "", false
+	}
+	return value, true
+}

